community/examples/gke-tpu-v4/gke-tpu-v4.yaml

```diff
@@ -0,0 +1,141 @@
+  modules:
+  - id: gke-tpu-v4-net
+    source: modules/network/vpc
+    settings:
+      network_name: $(vars.deployment_name)-net-0
+      mtu: 8896
+      subnetworks:
+      - subnet_name: $(vars.deployment_name)-sub-0
+        subnet_region: $(vars.region)
+        subnet_ip: 192.168.0.0/18
+      secondary_ranges_list:
+      - subnetwork_name: $(vars.deployment_name)-sub-0
+        ranges:
+        - range_name: pods
+          ip_cidr_range: 10.4.0.0/14
+        - range_name: services
+          ip_cidr_range: 10.0.32.0/20
+      firewall_rules:
+      - name: $(vars.deployment_name)-internal-0
+        ranges: [192.168.0.0/18]
+        allow:
+        - protocol: tcp
+          ports: ["0-65535"]
+        - protocol: udp
+          ports: ["0-65535"]
+        - protocol: icmp
+
+  - id: gke-tpu-v4-cluster
+    source: modules/scheduler/gke-cluster
+    use: [gke-tpu-v4-net]
+    settings:
+      system_node_pool_machine_type: "n1-standard-8"
+      system_node_pool_taints: []
+      enable_private_endpoint: false # Allows access from authorized public IPs
+      master_authorized_networks:
+      - cidr_block: $(vars.authorized_cidr) # Allows your machine to run the kubectl command. Required for multi network setup.
+        display_name: "kubectl-access-network"
+      # Cluster versions cannot be updated through the toolkit after creation
+      # Please manage cluster version from the Google Cloud Console directly
+      version_prefix: "1.32."
+      release_channel: RAPID
+      maintenance_exclusions:
+      - name: no-minor-or-node-upgrades-indefinite
+        start_time: "2024-12-01T00:00:00Z"
+        end_time: "2025-12-22T00:00:00Z"
+        exclusion_scope: NO_MINOR_OR_NODE_UPGRADES
+    outputs: [instructions]
+
+  - id: gke-tpu-v4-pool
+    source: modules/compute/gke-node-pool
+    use: [gke-tpu-v4-cluster]
+    settings:
+      num_slices: $(vars.num_slices)
+      name: gke-tpu-v4-pool
+      disk_type: pd-balanced
+      machine_type: ct4p-hightpu-4t # CT4P on cloud console? # causes an issue with a check
+      auto_upgrade: true
+      zones: [$(vars.zone)]
+      static_node_count: $(vars.static_node_count)
+      reservation_affinity:
+        consume_reservation_type: SPECIFIC_RESERVATION
+        specific_reservations:
+        - name: $(vars.extended_reservation)
+      placement_policy:
+        type: COMPACT
+        tpu_topology: $(vars.tpu_topology)
+    outputs: [instructions]
+
+  - id: job-template
+    source: modules/compute/gke-job-template
+    use: [gke-tpu-v4-pool]
+    settings:
+      image: ubuntu:latest
+      command:
+      - "/bin/bash"
+      - "-c"
+      - |
+        apt-get update && apt-get install -y python3 python3-venv python3-pip &&
+        python3 -m venv myenv &&
+        source myenv/bin/activate &&
+        pip install -U "jax[tpu]" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html &&
+        python3 -c 'import jax; print("TPU cores:", jax.device_count("tpu"))'
```

Comment:
This is not working. Debugging in process.

Reply:
No luck here. I will reach out for help.

Reply:
Should this be `jax.device_count()`?

Reply:
Any one of them can be used.

jax.device_count("tpu")  gives the total number of TPU cores in the Pod.